synapse/handlers/message.py

```diff
@@ -805,11 +811,15 @@ def handle_new_client_event(
             )
 
         try:
-            yield self.auth.check_from_context(room_version, event, context)
+            yield self.auth.check_from_context(room_version.identifier, event, context)
         except AuthError as err:
             logger.warning("Denying new event %r because %s", event, err)
             raise err
 
+        # Ensure the data is spec compliant JSON.
+        if room_version.strict_canonicaljson:
+            validate_canonicaljson(event.get_pdu_json())
```

Comment:
I was a little bit unsure if this should be `event.get_pdu_json()`. I originally had `event.content`, but it seems like we should be looking at the entire PDU.

Reply:
Seems sensible, the entire event should be canonicaljson if I understand the spec correctly.

Reply:
I switched this to `event.content`.